Saves power states and removes keybind modifiers.

SoaringEaqle opened the pull request:
## Changes proposed in this pull request:

Removes unappreciated modifier keybinds for individual activators

Adds power save states.

Power save states can be saved by holding the "save power save" key and pressing a hotbar slot key, while having at least one manifestation active.

Power save states can be activated by holding the "activate power save" key and pressing the hotbar key tied to the power save you wish to recall.

## Future Plans for These Features:
Chat message when power save state is saved or activated 
Save state toggle, turning those powers back off after use.

SoaringEaqle committed:
Saves power states and removes keybind modifiers.

Removes unappreciated modifier keybinds for individual activators

Adds power save states.

Power save states can be saved by holding the "save power save" key and pressing a hotbar slot key, while having at least one manifestation active.

Power save states can be activated by holding the "activate power save" key and pressing the hotbar key tied to the power save you wish to recall.

Future plans:
Chat message when power save state is saved or activated
Save state toggle, turning those powers back off after use.

leafreynolds commented:
Related discussion on discord
https://discord.com/channels/828383636465123328/1351674880831262860

SoaringEaqle commented:
Should resolve the following issues:
#119 
#106

SoaringEaqle committed:
Merge branch 'develop' into develop1

Gerbagel commented:
I really should have actually checked this ages ago, I'm *very* sorry.

Anyway, found a bug. 

1. Save a state
2. Load that state
3. Save a different state to the same slot
4. Load again, the state will unload instead
5. Attempt to load again, only some changes will be reflected. When testing, I set f-gold to store at 1, and a-steel to flare, and only a-steel changed while f-gold remained at 0

Also, I'm almost certain this is just me struggling with datagen again, but I'm only getting translation strings for each of the individual power hotkeys as well as the activate and save hotkeys. Could you confirm it works on your end?

SoaringEaqle commented:
I never did translations. I’ll tackle that with the bug fixes. I’ll also add a patchouli guide for navigating the keybinds.

SoaringEaqle committed:
Revert "Merge branch 'develop' into develop1"

This reverts commit 645e5422f5fb35fb3346a3e89b13de832d10fac0, reversing
changes made to 44934bd3f236cce77a8d765d9a12603cb596e1cc.

SoaringEaqle committed:
Reapply "Merge branch 'develop' into develop1"

This reverts commit e324649e49c8b383effe639aa148ae5f0b7e4bdd.

SoaringEaqle committed:
Bug Fixes, Patchouli Pages

Fixes bugs Gerbagel mentioned.
Adds translation keys for all new keybinds.
Separates dedicated activators into their own section.
Adds patchouli guide pages on key binds

SoaringEaqle committed:
Reformating

Signed-off-by: SoaringEaqle <158627775+SoaringEaqle@users.noreply.github.com>

SoaringEaqle commented:
Done with the bug fixes.

Gerbagel commented:
Works mostly as expected, though it still has to unload before loading a save state again. 

Looking at the code, you either need to set `isActive` to false when the state changes, or check if the current state is different from the state the player is attempting to load.

SoaringEaqle committed:
Bug Fixes

Fixes the issue with the toggle activation by checking the current mode of all manifestations.
Fixes a bug where save states would be deleted after quitting a world by saving a copy as nbt data attached to the SpiritwebCapability.
Adds a text pop-up to help show what is happening when saving and activating power save states.

SoaringEaqle commented:
I think I'm finally done. It might break on multiplayer, but I can't really test that.

leafreynolds commented:
> I think I'm finally done. It might break on multiplayer, but I can't really test that.

doesn't it let you runServer and runClient to connect to it?

SoaringEaqle commented:
I don't think I can test multiple players. The only reason I say it might break is because the power save states are copied to the sprirtweb as nbt data, and I have no clue how the server and client interact with that use case. It probably works, I just don't know how.

SoaringEaqle commented:
Upon further reflection and learning, I can say with 97% confidence that this will have zero issues on multiplayer.

Gerbagel commented:
No more issues that I can see. Multiplayer didn't crash, and it seems synced. Should be fine to merge if Leaf has nothing else.
